### PR TITLE
 Create a test suite for the Login/Signup page 

### PR DESCRIPTION
**Pull request by ghost**
This commit lays the groundwork for a pursuance test suite and adds a few simple tests for the login page. I've added the WebdriverIO test framework, the Chai assertion library, and some Babel support for writing ES6/ES2015 code.
```
$ npm run wdio

> pursuance@0.1.0 wdio /home/kwall/gocode/src/github.com/PursuanceProject/pursuance
> node -r babel-register ./node_modules/.bin/wdio

------------------------------------------------------------------
[firefox #0-0] Session ID: 17bf44f7-7e65-4efc-8a75-94cba4bed05d
[firefox #0-0] Spec: /home/kwall/gocode/src/github.com/PursuanceProject/pursuance/src/components/NavBar/LogIn/LogIn.test.js
[firefox #0-0] Running: firefox
[firefox #0-0]
[firefox #0-0] Pursuance login
[firefox #0-0]   ✓ should have a Log in link
[firefox #0-0]   ✓ should accept username input
[firefox #0-0]   ✓ should accept password input
[firefox #0-0]   ✓ should accept a valid username/password combination
[firefox #0-0]   !! should have a logout button
[firefox #0-0]   !! should reject an unknown username
[firefox #0-0]   !! should reject an invalid password
[firefox #0-0]
[firefox #0-0]
[firefox #0-0] 4 passing (4s)
[firefox #0-0] 3 pending
[firefox #0-0]
```
I imagine we'll have some conversations about this first. :)

The lint failure is not mine:
```
$ npm run lint
> pursuance@0.1.0 lint /home/kwall/gocode/src/github.com/PursuanceProject/pursuance
> npm run verify-eslint-rules && eslint --fix ./src && npm run format

> pursuance@0.1.0 verify-eslint-rules /home/kwall/gocode/src/github.com/PursuanceProject/pursuance
> eslint --print-config .eslintrc | eslint-config-prettier-check

No rules that are unnecessary or conflict with Prettier were found.

/home/kwall/gocode/src/github.com/PursuanceProject/pursuance/src/components/NavBar/NotificationsModal/NotificationsTab/NotificationsTab.js
  31:20  error  'FaSliders' is not defined  react/jsx-no-undef

✖ 1 problem (1 error, 0 warnings)

npm ERR! code ELIFECYCLE
```

### CODE REVIEW

File: src/components/CreatePursuance/CreatePursuance.test.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+describe('Creaing a pursuance', function() {
```
Comment (elimisteve) [inline src/components/CreatePursuance/CreatePursuance.test.js]: *Creating

**Comment by elimisteve** [comment]
@kurtwall Looks pretty great! Just make the tweak or 2 I commented about, and remove the `geckodriver` binary and do a force-push :+1: , then I'll do a final pass and we should be good.

File: src/components/CreatePursuance/CreatePursuance.test.js (hunk 1 above)
Comment (ghost) [reply src/components/CreatePursuance/CreatePursuance.test.js]: Fixed. Good catch.

**Comment by ghost** [comment]
Done and done. I'm excited to see my first commit go in!

**Comment by ghost** [comment]
@elimisteve Do you need anything more from me on this PR?

**Comment by elimisteve** [comment]
Sorry for the delay @kurtwall, we were at RightsCon in Toronto but now I'm back in action!

I ran through the instructions and for some reason I'm having trouble connecting the WebDriver client (IIUC?) to the Selenium server:

```
user@work1:~/gocode/src/github.com/PursuanceProject/pursuance$ npm run wdio wdio.conf.js

> pursuance@0.1.0 wdio /home/user/gocode/src/github.com/PursuanceProject/pursuance
> node -r babel-register ./node_modules/.bin/wdio "wdio.conf.js"

ERROR: connection refused
Build info: version: '3.12.0', revision: '7c6e0b3', time: '2018-05-08T15:15:08.936Z'
System info: host: 'work1', ip: '127.0.1.1', os.name: 'Linux', os.arch: 'amd64', os.version: '4.9.56-21.pvops.qubes.x86_64', java.version: '1.8.0_171'
Driver info: driver.version: unknown
remote stacktrace: 
firefox
    at new RuntimeError (/home/user/gocode/src/github.com/PursuanceProject/pursuance/node_modules/webdriverio/build/lib/utils/ErrorHandler.js:144:12)
    at Request._callback (/home/user/gocode/src/github.com/PursuanceProject/pursuance/node_modules/webdriverio/build/lib/utils/RequestHandler.js:316:39)
    at Request.self.callback (/home/user/gocode/src/github.com/PursuanceProject/pursuance/node_modules/request/request.js:186:22)
    at emitTwo (events.js:106:13)
    at Request.emit (events.js:194:7)
    at Request.<anonymous> (/home/user/gocode/src/github.com/PursuanceProject/pursuance/node_modules/request/request.js:1163:10)
    at emitOne (events.js:96:13)
    at Request.emit (events.js:191:7)
    at IncomingMessage.<anonymous> (/home/user/gocode/src/github.com/PursuanceProject/pursuance/node_modules/request/request.js:1085:12)
    at Object.onceWrapper (events.js:293:19)
```

**Comment by elimisteve** [comment]
Java output right when it starts up:

```
user@work1:~/gocode/src/github.com/PursuanceProject/pursuance$ java -jar -Dwebdriver.gecko.driver=./geckodriver ./selenium-server-standalone-3.12.0.jar
23:27:05.725 INFO [GridLauncherV3.launch] - Selenium build info: version: '3.12.0', revision: '7c6e0b3'
23:27:05.727 INFO [GridLauncherV3$1.launch] - Launching a standalone Selenium Server on port 4444
2018-05-23 23:27:05.910:INFO::main: Logging initialized @737ms to org.seleniumhq.jetty9.util.log.StdErrLog
23:27:06.168 INFO [SeleniumServer.boot] - Selenium Server is up and running on port 4444
23:27:44.005 INFO [ActiveSessionFactory.apply] - Capabilities are: Capabilities {browserName: firefox, handlesAlerts: true, javascriptEnabled: true, locationContextEnabled: true, loggingPrefs: org.openqa.selenium.logging..., maxInstances: 5, requestOrigins: {name: webdriverio, url: http://webdriver.io, version: 4.12.0}, rotatable: true}
23:27:44.007 INFO [ActiveSessionFactory.lambda$apply$11] - Matched factory org.openqa.selenium.remote.server.ServicedSession$Factory (provider: org.openqa.selenium.firefox.GeckoDriverService)
1527143264050	geckodriver	INFO	geckodriver 0.20.1
1527143264078	geckodriver	INFO	Listening on 127.0.0.1:24829
1527143264532	mozrunner::runner	INFO	Running command: "/usr/bin/firefox" "-marionette" "-profile" "/tmp/rust_mozprofile.HU6IO7h2hPa6"
```

**Comment by elimisteve** [comment]
@kurtwall When Firefox starts up there's no URL in the URL bar, which seems a little odd to me, but I'm not sure.

**Comment by elimisteve** [comment]
@kurtwall Any tips on getting the ~client to connect to the Selenium server?

**Comment by elimisteve** [comment]
@kurtwall This looks good but I had issues getting the tests to run on my machine. Any tips? I can look into this, too.

**Comment by ghost** [comment]
@elimisteve O hai!
Been a little hairy here since the beginning of June. I'll look into this tonight and address your review comments.

**Pull request closed without merging**